llmfoundry/utils/exceptions.py

```diff
@@ -79,13 +79,21 @@ def __init__(
 class UnknownExampleTypeError(KeyError):
     """Error thrown when an unknown example type is used in a task."""
 
-    def __init__(self, example: Mapping) -> None:
+    def __init__(self, example: Union[Mapping, str]) -> None:
```

Comment:
@milocress Do you understand why this can sometimes be a string? I'm confused because we only raise this error from a single point in the code where the type of example is `Example`, so something must be wrong with our typing expectations at the call-site.

Reply:
It's odd too because the function that calls this explicitly errors if `example` is not a `Mapping`. I suspect it has something to do with threading and pickling.

Reply:
example seems to be a `KeysView` object:

```
KeysView({'prompt': 'hello, ', 'response': 'world!', 'random_extra_key': 'sup'})
```

Reply:
```
    if not isinstance(example, Mapping):
        raise TypeError(
            f'Expected example to be a Mapping, but found {type(example)}',
        )
```
Given that we make this check here, I have no clue why this can sometimes be a string, beyond some kind of threading/race condition issue. I think there's something spooky happening